misc/dbt-materialize/dbt/include/materialize/macros/deploy/deploy_await.sql

```diff
@@ -38,7 +38,7 @@
 {% set clusters = target_config.get('clusters', []) %}
 
 {% for cluster in clusters %}
-    {% set deploy_cluster = cluster ~ "_dbt_deploy" %}
+    {% set deploy_cluster = generate_cluster_name_internal(cluster, force_deploy_suffix=True) %}
```

Comment:
By using the `generate_cluster_name_internal` macro in the deploy macros instead of directly concatenating `cluster ~ "_dbt_deploy"`, we enable users to override the `generate_cluster_name` macro without disrupting the deploy workflow.

For example, if a user defines their own `generate_cluster_name` macro like this:

```jinja
{% macro materialize__generate_cluster_name(custom_cluster_name, node) -%}
    {%- set default_cluster = target.cluster -%}
    {%- if custom_cluster_name is none -%}
        {{ default_cluster }}
    {%- else -%}
        {{ target.name }}_{{ custom_cluster_name }}
    {%- endif -%}
{%- endmacro %}
```

The blue/green deployments will remain compatible with this custom logic. The deployment workflow will correctly generate names such as `dev_some_name_dbt_deploy`, where `dev` is the user-defined target name.